vault/src/lib.rs

```diff
@@ -205,6 +206,22 @@ pub async fn start<B: BitcoinCoreApi + Send + Sync + 'static>(
         } else {
             info!("Not registering vault -- already registered");
         }
+    } else {
+        match arc_provider.get_vault(vault_id.clone()).await {
```

Comment:
Probably you should also do this check in the two cases above? And if you refactor anyway, maybe you find a way to remove the `info!` duplication above?

Reply:
Moved the checks to the RPC interface, I'm not worried about the duplication since most of the faucet stuff will be removed prior to Kusama.